src/Illuminate/Database/Eloquent/Relations/Relation.php

```diff
@@ -283,7 +283,7 @@ protected function getKeys(array $models, $key = null)
     {
         return collect($models)->map(function ($value) use ($key) {
             return $key ? $value->getAttribute($key) : $value->getKey();
-        })->values()->unique(null, true)->sort()->all();
+        })->values()->unique(null, true)->filter()->sort()->all();
```

Comment:
0 is a legal PK value. Empty string either.

Reply:
Thanks! I've updated to explicitly check null now.